supervisor/mounts/mount.py

```diff
@@ -273,6 +291,8 @@ async def unmount(self) -> None:
                 await self.sys_dbus.systemd.reset_failed_unit(self.unit_name)
             else:
                 await self.sys_dbus.systemd.stop_unit(self.unit_name, StopUnitMode.FAIL)
+
+            await self._update_state_await(UnitActiveState.INACTIVE)
```

Comment:
It may not go to `INACTIVE`, it could also fail. We really want to know when it has transitioned to any state that is considered stopped.

Reply:
Yeah I was thinking that we would just timeout, no big deal :man_shrugging:

> We really want to know when it has transitioned to any state that is considered stopped.

That begs the question what states there are. Systemd sources list these:

https://github.com/systemd/systemd/blob/v249.12/src/basic/unit-def.c#L99-L107

There is also "maintenance" and "reloading" :thinking: 

We should probably rely on these high level categorizations:
https://github.com/systemd/systemd/blob/v249.12/src/core/unit.h#L38-L52

But yeah, instead of timing out, we really can do it smarter rather easily, so why not. Updated the code to also wait for "failed" state.